src/fromager/commands/build.py

```diff
@@ -278,6 +284,15 @@ def _summary(ctx: context.WorkContext, entries: list[BuildSequenceEntry]) -> Non
         console = rich.console.Console(file=f, width=sys.maxsize)
         console.print(*output, sep="\n\n")
 
+    with open(ctx.work_dir / "build-sequence-summary.json", "w", encoding="utf-8") as f:
+        json.dump(
+            [
+                dataclasses.asdict(e, dict_factory=BuildSequenceEntry.dict_factory)
```

Comment:
I _think_ this could just be `dataclasses.asdict(e)`. Did you find you needed to pass the factory function explicitly or does that do something extra?

Reply:
So both Version and pathlib.Path failed to serialize with errors like

```
ERROR: Object of type PosixPath is not JSON serializable     
```

Using the dict_factory here to use __str__ for those worked. I debated switching to pydantic here, used else but this seemed sufficient given it's a report.

Reply:
What you have looks good. Pydantic wouldn't be wrong, but is probably an extra complication here.